plots and metrics: update new targets rules (#1809)
iterative/dvc.org: content/docs/command-reference/plots/show.md

                      [targets [targets ...]]

positional arguments:

>   targets               Files to visualize (supports any file,

Reviewer: sounds too explicit. Why do we change this. I think if explain in the `plots` concept itself that plots are not only `dvc.yaml` based people won't expect `dvc.yaml` to be present in the first.

Reviewer 2: Oops. I approved this change in the core repo: https://github.com/iterative/dvc/pull/4620#pullrequestreview-497717870

Reviewer 2: > Files to visualize (supports any file, even when not found as `plots` in `dvc.yaml`). Shows all plots by default.

So how about changing/expanding on "All plots defined in `dvc.yaml` are used by default." in the **Description**, and simplifying the above to just "Plot files to visualize (shows all plots defined in `dvc.yaml` by default)" ?

Reviewer: yep, sounds good to me ... still a bit verbose, and `dvc.yaml` will be confusing people but I don't know how to do it better

Author: Updated the description for plots `show` and `diff`, `metrics` seem alright. Please ping me if its ok as is now, I'll prepare PR for help messages to `dvc`.

Author: @shcheklein, do you think its ok now?

                      [targets [targets ...]]

positional arguments:

>   targets              Plot files to visualize (shows all plots defined in dvc.yaml by default).

Reviewer 2: Wrapping at 72 chars
```suggestion
  targets     Plot files to visualize (shows all plots defined in
              dvc.yaml by default).
```

                      [targets [targets ...]]

positional arguments:

>   targets     Plot files to visualize (shows all plots defined in

Reviewer 2: ```suggestion
  targets     Plots files to visualize (shows all plots defined in
```

AUC curves, confusion matrices, etc. All plots defined in `dvc.yaml` are used by
default.

> Optionally, specific metric file `targets` to show are accepted. They do not not
> have to be specified as `plots` in `dvc.yaml`.

Reviewer 2: ```suggestion
Optionally, specific metrics file `targets` to show are accepted  (supports any
valid plots, even when not found as `plots` in `dvc.yaml`, see the `--plots`
option of `dvc run`).
```

> Optionally, specific metric file `targets` to show are accepted. If `targets`
> are present, they do not have to be defined in `dvc.yaml` as plots.

Reviewer 2: ```suggestion
Optionally, specific metric file `targets` to show are accepted. Note that these
don't have to be defined as `plots` in `dvc.yaml` necessarily.
```

Reviewer 2: May break the format

> Optionally, specific metrics file `targets` to show are accepted. If `targets`
> are present, they do not have to be defined in `dvc.yaml` as plots.

Reviewer 2: ```suggestion
Optionally, specific metric file `targets` to show are accepted. Note that these
don't have to be defined as `plots` in `dvc.yaml`.
```

AUC curves, confusion matrices, etc. All plots defined in `dvc.yaml` are used by
default.

> Optionally, specific metrics file `targets` to show are accepted  (supports any

Reviewer 2: ```suggestion
Optionally, specific metrics file `targets` to show are accepted (supports any
```
